apps/myaccount/src/hooks/use-sign-in.ts

```diff
@@ -0,0 +1,441 @@
+
+                // If `authorize` endpoint is overridden, save that in the session.
+                if (window["AppUtils"].getConfig().idpConfigs?.authorizeEndpointURL) {
+                    authorizationEndpoint = resolveIdpURLSAfterTenantResolves(
+                        authorizationEndpoint,
+                        window["AppUtils"].getConfig().idpConfigs.authorizeEndpointURL
+                    );
+                }
+
+                // If super tenant proxy is configured, `authorize` endpoint is updated with the configured
+                // super tenant proxy.
+                if (window["AppUtils"].getConfig().superTenantProxy) {
+                    authorizationEndpoint = authorizationEndpoint.replace(
+                        window["AppUtils"].getConfig().superTenant,
+                        window["AppUtils"].getConfig().superTenantProxy
+                    );
+                }
+
+                // If `oidc session iframe` endpoint is overridden, save that in the session.
+                if (window["AppUtils"].getConfig().idpConfigs?.oidcSessionIFrameEndpointURL) {
+                    oidcSessionIframeEndpoint = resolveIdpURLSAfterTenantResolves(
+                        oidcSessionIframeEndpoint,
+                        window["AppUtils"].getConfig().idpConfigs.oidcSessionIFrameEndpointURL
+                    );
+                }
+
+                // If `token` endpoint is overridden, save that in the session.
+                if (window["AppUtils"].getConfig().idpConfigs?.tokenEndpointURL) {
+                    tokenEndpoint = resolveIdpURLSAfterTenantResolves(
+                        tokenEndpoint,
+                        window["AppUtils"].getConfig().idpConfigs.tokenEndpointURL
+                    );
+                }
+
+                sessionStorage.setItem(AUTHORIZATION_ENDPOINT, authorizationEndpoint);
+                sessionStorage.setItem(OIDC_SESSION_IFRAME_ENDPOINT, oidcSessionIframeEndpoint);
+                sessionStorage.setItem(TOKEN_ENDPOINT, tokenEndpoint);
+
+                // Update post_logout_redirect_uri of logout_url with tenant qualified url
+                const logoutUrl: string = window["AppUtils"].getConfig().idpConfigs?.logoutEndpointURL;
+                let logoutRedirectUrl: string;
+
+                updateConfig({
+                    endpoints: {
+                        authorizationEndpoint: authorizationEndpoint,
+                        checkSessionIframe: oidcSessionIframeEndpoint,
+                        endSessionEndpoint: logoutUrl.split("?")[0],
+                        tokenEndpoint: tokenEndpoint
+                    },
+                    signOutRedirectURL: deriveLogoutRedirectForSubOrgLogins(
+                        logoutRedirectUrl,
+                        idToken.user_org,
+                        idToken.org_id
+                    )
+                });
+            })
+            .catch((error: AsgardeoAuthException) => {
```

Comment:
We can use IdentityAppsError instead, I think.

Reply:
`getDecodedIDToken` is from `@asgardeo/auth-react` and throws `AsgardeoAuthException`.
If we are throwing a new `IdentityAppsError` we may have to create a `new  IdentityAppsError` and throw that instead.

Was that the expectation?